### PR TITLE
Support symbolic arrays in `@mtkmodel`

### PR DESCRIPTION
**Pull request by ven-k**
- Adds symbolic arrays to `@mtkmodel`
- Adds relevant tests and docs
- Updates syntax of  `@extend`  in the docs

### CODE REVIEW

**Comment by ven-k** [comment]
The test failure is unrelated to this PR; I checked that even master throws the same error.

**Comment by ven-k** [comment]
Model Parsing tests pass  for Julia 1.6[^1]

[^1]:https://github.com/SciML/ModelingToolkit.jl/actions/runs/6611821413/job/17956475971?pr=2282

File: src/systems/model_parsing.jl [hunk 1]
```diff
@@ -176,7 +179,12 @@ function generate_var!(dict, a, b, varclass)
         Dict{Symbol, Dict{Symbol, Any}}()
     end
     vd[a] = Dict{Symbol, Any}()
-    var = Symbolics.variable(a, T = SymbolicUtils.FnType{Tuple{Real}, Real})(iv)
+    var = if indices === nothing
+        Symbolics.variable(a, T = SymbolicUtils.FnType{Tuple{Real}, Real})(iv)
+    else
+        vd[a][:length] = lastindex(indices)
```
Comment (YingboMa) [inline src/systems/model_parsing.jl:185]: This won't general arrays. Shouldn't it be `size`?

**Comment by ven-k** [comment]
Bump

**Review by YingboMa — APPROVED** [review]

**Pull request merged**